src/Esfa.Ofsted.Inspection.Client/Services/GetOfstedDetailsFromExcelPackageService.cs

```diff
@@ -110,26 +117,58 @@ public InspectionOutcomesResponse ExtractOfstedInspections(ExcelPackage package)
                 return InspectionsStatusCode.Success;
             }
 
-            error.Website = url;
+            error.Website = url ?? string.Empty;
+
             errorSet.Add(error);
-            _logger.Warn($"Details processed unsuccessfully for line {error.LineNumber}: '{error.LineNumber}', '{error.Website}', '{error.DatePublished}', '{error.OverallEffectiveness}'");
+            _logger.Warn($"Details processed unsuccessfully for line {error.LineNumber}: '{error.Ukprn}', '{error.Website}', '{error.DatePublished}', '{error.OverallEffectiveness}'");
             return InspectionsStatusCode.ProcessedWithErrors;
 
         }
 
-
-        private static int FindStartingLineNumber(ExcelWorksheet keyWorksheet)
+        private SpreadsheetDetails GetSpreadsheetColumnAndRowDetails(ExcelWorksheet keyWorksheet)
         {
+            var spreadsheetDetails = new SpreadsheetDetails();
+            var matchFound = false;
             var lineNumberStart = keyWorksheet.Dimension.Start.Row;
-            int intValue;
-            while (!int.TryParse(keyWorksheet.Cells[lineNumberStart, UkprnPosition]?.Value?.ToString(), out intValue) &&
-                   GetDateTimeValue(keyWorksheet.Cells[lineNumberStart, DatePublishedPosition]) == null && lineNumberStart<= keyWorksheet.Dimension.End.Row)
+            while (!matchFound && lineNumberStart <= keyWorksheet.Dimension.End.Row)
             {
+                for (var i = 1; i <= keyWorksheet.Dimension.End.Column; i++)
+                {
+                    if (keyWorksheet.Cells[lineNumberStart, i]?.Value?.ToString() == _configurationSettings.WebLinkHeading)
+                    {
```

Comment:
may be we can replace getting the value with a variable 
`var cellValue = keyWorksheet.Cells[lineNumberStart, i]?.Value?.ToString().Trim()` ? 

and may be we can use 
```
if (string.Equals(cellValue, _configurationSettings.WebLinkHeading, StringComparison.OrdinalIgnoreCase))
{
  spreadsheetDetails.WebLinkColumn = i;
  matchFound = true;
}
```